it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyControllerISpec.scala

```diff
@@ -51,6 +53,93 @@ trait CheckAgencyControllerISpec extends BaseISpec with SessionDataMissingSpec {
 
   lazy val controller: CheckAgencyController = app.injector.instanceOf[CheckAgencyController]
 
+  "showCheckBusinessType (GET /check-business-type)" should {
+    behave like anAgentAffinityGroupOnlyEndpoint(controller.showCheckBusinessType(_))
+
+    behave like aPageTakingContinueUrlAndCachingInSessionStore(controller.showCheckBusinessType(_),
+      sessionStoreService, userIsAuthenticated(subscribingCleanAgentWithoutEnrolments))
+
+    "contain page titles and header content" in {
+      val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+      val result = await(controller.showCheckBusinessType(request))
+
+      checkHtmlResultWithBodyText(result,
+        htmlEscapedMessage("checkBusinessType.title"),
+        htmlEscapedMessage("checkBusinessType.progressive.title"),
+        htmlEscapedMessage("checkBusinessType.progressive.content.p1")
+      )
+    }
+
+    "contain radio options for Sole Trader, Limited Company, Partnership, and LLP" in {
+      val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+      val result = await(controller.showCheckBusinessType(request))
+      val doc = Jsoup.parse(bodyOf(result))
+
+      // Check form's radio inputs have correct values
+      doc.getElementById("businessType-soletrader").`val`() shouldBe "soletrader"
+      doc.getElementById("businessType-limited").`val`() shouldBe "limited"
+      doc.getElementById("businessType-partnership").`val`() shouldBe "partnership"
+      doc.getElementById("businessType-llp").`val`() shouldBe "llp"
+    }
+
+    "contain a link to sign out" in {
+      val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+      val result = await(controller.showCheckBusinessType(request))
+      val doc = Jsoup.parse(bodyOf(result))
+      val signOutLink = doc.getElementById("sign-out")
+      signOutLink.attr("href") shouldBe routes.SignedOutController.signOutWithContinueUrl.url
+      signOutLink.text() shouldBe htmlEscapedMessage("checkBusinessType.progressive.content.link")
+    }
+  }
+
+  "submitCheckBusinessType (POST /check-business-type)" when {
```

Comment:
there are 4 almost identical tests for different form input, lets optimise that and use behave like.

Reply:
Good call